src/theory/theory_engine.cpp

```diff
@@ -1451,6 +1451,12 @@ void TheoryEngine::lemma(TrustNode tlemma,
   // assert the lemma
   d_propEngine->assertLemma(tlemma, p);
 
+  // We have asserted this lemma, so we collect it in partition generator.
```

Comment:
This should be part of `PartitionGenerator::notifyLemma`.

On a related note, it seems that the check on line 1463 was previously wrong (I had fixed this on my branch https://github.com/ajreynol/CVC4/tree/plugin, but this was not merged to main).

Can you change the condition `if (d_relManager != nullptr)` to `if (!d_modules.empty())`?

Reply:
Fixed both.